Sources/Context/Context.swift

```diff
@@ -0,0 +1,54 @@
+public protocol Context {
+    static var background: Context { get }
+    static func TODO(_ reason: StaticString?, function: StaticString, file: StaticString, line: UInt) -> Context
+    
+    subscript<Key: ContextValueContainerKey>(_ key: Key.Type) -> Key.Value? { get set }
+    
+    func forEach(_ body: (AnyContextValueContainerKey, Any) throws -> Void) rethrows
```

Comment:
This signature is very confusing on context itself.

Say there's vapor's request and it conforms to context:

```swift
request.forEach { key, value in ... }
```

This is super confusing!

This is why the final proposal design, while `Context` still was conforming to `BaggageProtocol` did have this function called `forEachBaggageItem` -- which also follows prior art in existing tracing APIs. We should not make confusingly named APIs like `foreach` and a subscript like that available on any context just so.